src/Symfony/Component/Form/Form.php

```diff
@@ -374,10 +379,12 @@ public function setData($modelData)
             // Update child forms from the data (unless their config data is locked)
             $this->config->getDataMapper()->mapDataToForms($viewData, new \RecursiveIteratorIterator(new InheritDataAwareIterator($this->children)));
         }
-
-        if ($dispatcher->hasListeners(FormEvents::POST_SET_DATA)) {
-            $event = new PostSetDataEvent($this, $modelData);
-            $dispatcher->dispatch($event, FormEvents::POST_SET_DATA);
+        foreach ($thisModelDataAwareForms as $form) {
```

Comment:
I don't think this should reuse the same array. The form might have been modified in the meantime.

Reply:
You're right! I didn't think that some `PRE_SET_DATA` listeners may also alter form's structure.
What should I do in this case? Use `getSameModelAwareForms` at every iteration and exclude form elements which I already dispatched the event on to?

Reply:
Now I'm using iterators 👍🏻